test/setup/main.tf

```diff
@@ -32,5 +32,34 @@ module "project" {
     "servicenetworking.googleapis.com",
     "cloudkms.googleapis.com",
     "alloydb.googleapis.com",
+    "servicedirectory.googleapis.com",
+    "dns.googleapis.com",
   ]
 }
+
+module "psc_project" {
+  source  = "terraform-google-modules/project-factory/google"
+  version = "~> 15.0"
+
+  name              = "ci-alloy-db-psc"
+  random_project_id = "true"
+  org_id            = var.org_id
+  folder_id         = var.folder_id
+  billing_account   = var.billing_account
+
+  // get the required APIs for alloydb
+  activate_apis = [
+    "cloudresourcemanager.googleapis.com",
+    "storage-api.googleapis.com",
+    "serviceusage.googleapis.com",
+    "servicenetworking.googleapis.com",
+    "cloudkms.googleapis.com",
+    "alloydb.googleapis.com",
+    "servicedirectory.googleapis.com",
+    "dns.googleapis.com",
+  ]
+}
+
+data "google_project" "project" {
```

Comment:
nit: you can directly get this via https://github.com/terraform-google-modules/terraform-google-project-factory/blob/6e5e38827df1f2fc84f41ad746e3ede44bcd1db3/outputs.tf#L27